pkg/init/cmd/service/runc.go

```diff
@@ -103,9 +136,33 @@ func runcInit(rootPath string) int {
 
 		cleanup(path)
 		_ = os.Remove(pidfile)
+
+		// dump the log file outputs to os.Stdout/os.Stderr
+		if _, err = stdout.Seek(0, 0); err != nil {
+			log.Printf("Error resetting stdout of onboot service %s to console: %v", name, err)
+		} else if _, err := io.Copy(os.Stdout, stdout); err != nil {
+			log.Printf("Error writing stdout of onboot service %s to console: %v", name, err)
+		}
+
+		if _, err = stderr.Seek(0, 0); err != nil {
+			log.Printf("Error resetting stderr of onboot service %s to console: %v", name, err)
+		} else if _, err := io.Copy(os.Stderr, stderr); err != nil {
+			log.Printf("Error writing stderr of onboot service %s to console: %v", name, err)
+		}
+
 	}
 
 	_ = os.RemoveAll(tmpdir)
 
+	// make sure the link exists from /var/log/onboot -> /run/log/onboot
+	if err := os.MkdirAll(varLogDir, 0755); err != nil {
+		log.Printf("Error creating secondary log directory %s: %v", varLogDir, err)
+	} else if _, err := os.Stat(varLogLink); os.IsNotExist(err) {
+		// create the link
+		if err := os.Symlink(logDir, varLogLink); err != nil {
```

Comment:
Not a big deal, but the normal in Go would be to just call `os.Symlink` and squash an `os.IsExist` result that than doing the stat.

Reply:
Oh ok. That makes sense. I will do that.

> Not a big deal,

No, it isn't but clean code is better.

Reply:
Actually, I don't see how that would help. I would need to handle 2 error conditions: `os.IsExist` and anything else. So my code would be:

```go
	if err := os.MkdirAll(varLogDir, 0755); err != nil {
		log.Printf("Error creating secondary log directory %s: %v", varLogDir, err)
	} else {
                _, err := os.Symlink(logDir, varLogLink)
                switch {
                   case os.IsExist(err):
                        // do nothing, maybe report it
			log.Printf("Cannot create symlink from %s to %s, as file already exists", varLogLink, logDir)
                    default:
			log.Printf("Error creating symlink from %s to %s: %v", varLogLink, logDir, err)
		}
	}
```

Is that any better?

Reply:
No, I got it. It is cleaner what you suggested. 

```go
	// make sure the link exists from /var/log/onboot -> /run/log/onboot
	if err := os.MkdirAll(varLogDir, 0755); err != nil {
		log.Printf("Error creating secondary log directory %s: %v", varLogDir, err)
	} else if err := os.Symlink(logDir, varLogLink); err != nil && !os.IsExist(err) {
			log.Printf("Error creating symlink from %s to %s: %v", varLogLink, logDir, err)
		}
	}
```

Reply:
Yep, that's what I was envisaging (apart from the indentation)

Reply:
Took a try or two, but we got it. :-)

If you are good with the rest @ijc , please hit the approve button. Since you and Justin both commented, want approval from both (he already did).